src/library_fs.js

```diff
@@ -199,8 +199,14 @@ FS.staticInit();
           break;
         }
 
-        current = FS.lookupNode(current, parts[i]);
         current_path = PATH.join2(current_path, parts[i]);
+        try {
+          current = FS.lookupNode(current, parts[i]);
+        } catch (e) {
+          if ((e?.errno === {{{ cDefs.ENOENT }}}) && islast && opts.handleBrokenLink) {
+            return { path: current_path };
+          }
+        }
```

Comment:
Makes me a little sad to increase the complexity of `lookupPath`.  I already have no trouble understanding it.   But I get that it might be needed in this case :(

Reply:
Yeah I have trouble understanding it too. I think it would help to remove the weird mixture of recursion and iteration. I think it would be possible to hoist the innermost while loop up to the top of the function and remove the recursive call. But it would be useful if this function had detailed unit tests for that.

Reply:
Okay I think I successfully refactored `lookupPath` to not make recursive calls. I will open a separate PR for that.